src/routing_membrane.rs

```diff
@@ -340,6 +352,36 @@ impl<F> RoutingMembrane<F> where F: Interface {
         // TODO: trigger churn on boolean
     }
 
+    /// Parse and update the header with us as a relay node;
```

Comment:
I think we shouldn't wrap the `name` in the `Relay` enum at line 206 and then check whether it really is wrapped in `Relay` here. Instead, the `received_from` should have directly the type `NameType`.